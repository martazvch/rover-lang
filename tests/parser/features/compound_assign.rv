a += 4
a -= 7
a *= 5
a /= 4

