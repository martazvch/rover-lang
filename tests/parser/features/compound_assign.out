{
    "assignment": {
        "assigne": {
            "literal": "a"
        },
        "value": {
            "binop": {
                "lhs": {
                    "literal": "a"
                },
                "rhs": {
                    "literal": "4"
                },
                "op": "+"
            }
        }
    },
    "assignment": {
        "assigne": {
            "literal": "a"
        },
        "value": {
            "binop": {
                "lhs": {
                    "literal": "a"
                },
                "rhs": {
                    "literal": "7"
                },
                "op": "-"
            }
        }
    },
    "assignment": {
        "assigne": {
            "literal": "a"
        },
        "value": {
            "binop": {
                "lhs": {
                    "literal": "a"
                },
                "rhs": {
                    "literal": "5"
                },
                "op": "*"
            }
        }
    },
    "assignment": {
        "assigne": {
            "literal": "a"
        },
        "value": {
            "binop": {
                "lhs": {
                    "literal": "a"
                },
                "rhs": {
                    "literal": "4"
                },
                "op": "/"
            }
        }
    }
}
